src/server/scripts/Northrend/VioletHold/boss_erekem.cpp

```diff
@@ -219,21 +186,115 @@ class boss_erekem : public CreatureScript
+                    if (Creature* pGuard2 = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_EREKEM_GUARD_2)))
+                    {
+                        if (!pGuard1->IsAlive() && !pGuard2->IsAlive())
+                        {
+                            phase = 1;
+                            DoCastVictim(SPELL_STORMSTRIKE);
+                            DoCast(SPELL_WINDFURY);
+                            events.Reset();
+                            events.ScheduleEvent(EVENT_EARTH_SHOCK, urand(2000, 8000));
+                            events.ScheduleEvent(EVENT_WINDFURY, urand(1500, 2000));
+                            events.ScheduleEvent(EVENT_STORMSTRIKE, urand(1500, 2000));
+                        }
+                    }
+                }
+
+            events.Update(diff);
 
-            Creature* pGuard2 = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_EREKEM_GUARD_2));
-            if (pGuard2 && pGuard2->IsAlive() && !pGuard2->HealthAbovePct(75))
-                return pGuard2->GetGUID();
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
 
-            return ObjectGuid::Empty;
+            if (breakBondsCd <= 0)
+            {
+                if (Creature* pGuard1 = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_EREKEM_GUARD_1)))
+                {
+                    if (Creature* pGuard2 = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_EREKEM_GUARD_2)))
+                    {
+                        if (pGuard1->IsAlive())
+                        {
+                            if (pGuard1->HasAuraType(SPELL_AURA_MOD_STUN) || pGuard1->HasAuraType(SPELL_AURA_MOD_ROOT)
+                                || pGuard1->HasAuraType(SPELL_AURA_MOD_CONFUSE) || pGuard1->HasAuraType(SPELL_AURA_MOD_PACIFY)
+                                || pGuard1->HasAuraType(SPELL_AURA_MOD_DECREASE_SPEED))
+                            {
+                                DoCast(SPELL_BREAK_BONDS);
+                                breakBondsCd = 10000;
+                                return;
+                            }
+                        }
+                        if (pGuard2->IsAlive())
+                        {
+                            if (pGuard2->HasAuraType(SPELL_AURA_MOD_STUN) || pGuard2->HasAuraType(SPELL_AURA_MOD_ROOT)
+                                || pGuard2->HasAuraType(SPELL_AURA_MOD_CONFUSE) || pGuard2->HasAuraType(SPELL_AURA_MOD_PACIFY)
+                                || pGuard2->HasAuraType(SPELL_AURA_MOD_DECREASE_SPEED))
+                            {
+                                DoCast(SPELL_BREAK_BONDS);
+                                breakBondsCd = 10000;
+                                return;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+                breakBondsCd -= diff;
+
+            switch (uint32 eventId = events.ExecuteEvent())
```

Comment:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_erekem.cpp: In member function ‘virtual void boss_erekem::boss_erekemAI::UpdateAI(uint32)’:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_erekem.cpp:257:28: warning: unused variable ‘eventId’ [-Wunused-variable]
             switch (uint32 eventId = events.ExecuteEvent())
                            ^